src/utlist.h

```diff
@@ -349,6 +349,26 @@ do {
   }                                                                                            \
 } while (0)
 
+#define LL_INSERT_INORDER(head,add,cmp)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
```

Comment:
Weird, GitHub ate one of my line comments. I'll repeat it: I suspect that this new code won't work with `NO_DECLTYPE`. Can you investigate it a little bit (e.g. by writing some test cases, and then trying to compile those test cases with `-DNO_DECLTYPE=1`) and fix any really low-hanging fruit?
I'm not too worried if there remains fruit that's not low-hanging. Worst case, some list macros will be unusable on compilers with `NO_DECLTYPE`; but then people on those compilers can just avoid those macros and they'll be fine.